lib/utils/file_uploader_util.dart

```diff
@@ -140,15 +142,31 @@ Future<MediaUploadData> _getMediaUploadDataFromAppCache(ente.File file) async {
     _logger.warning("File doesn't exist in app sandbox");
     throw InvalidFileError("File doesn't exist in app sandbox");
   }
-  thumbnailData = await getThumbnailFromInAppCacheFile(file);
-  return MediaUploadData(sourceFile, thumbnailData, isDeleted);
+  try {
+    thumbnailData = await getThumbnailFromInAppCacheFile(file);
+    return MediaUploadData(sourceFile, thumbnailData, isDeleted);
+  } catch (e, s) {
+    _logger.severe("failed to generate thumbnail", e, s);
+    throw InvalidFileError(
+        "thumbnail generation failed for fileType: ${file.fileType.toString()}");
+  }
 }
 
 Future<Uint8List> getThumbnailFromInAppCacheFile(ente.File file) async {
   var localFile = io.File(getSharedMediaFilePath(file));
   if (!localFile.existsSync()) {
     return null;
   }
+  if (file.fileType == FileType.video) {
+    final thumbnailFilePath = await VideoThumbnail.thumbnailFile(
+      video: localFile.path,
+      imageFormat: ImageFormat.JPEG,
+      thumbnailPath: (await getTemporaryDirectory()).path,
+      maxWidth: kThumbnailLargeSize,
+      quality: 100,
```

Comment:
`80` should be sufficient.

Reply:
https://sirv.sirv.com/Examples/test-girl.jpg?quality=100 vs https://sirv.sirv.com/Examples/test-girl.jpg?quality=80

There won't be any discernible difference.